sig-architecture/generics.md

```diff
@@ -0,0 +1,22 @@
+# Generics Policy
+
+Generics may be used in Kubernetes starting in v1.25
+
+In Kubernetes v1.24 we shipped Go 1.18 (which adds support for generics)
+rather late in the release cycle, so we temporarily prohibited using generics,
+in case we ran into other issues and needed to roll back to unblock the release.
+Now that v1.24.0 is out, Kubernetes generics should be unrestricted.
+
+In general, Kubernetes does not restrict using new Go features, we've quickly
+adopted new standard library types and methods and will continue to do so.
+
+Generally the latest stable go release is in use on the main development branch.
+This includes all of the staging libraries (client-go etc.) that originate in the
+main Kubernetes/Kubernetes repository.
+
+## Recommendations for Reviewers
+
+Consider if proposed generics pull requests improve maintainability and readability.
+
+The current generics implementation is known to have some performance issues
+depending on usage: consider requesting benchmarks before / after the changes.
```

Comment:
Also...

Avoid generics If this is a bug fix in master branch which is likely to be back ported into stable branches that are not on go 1.18.

Reply:
added